demo/src/app/components/typeahead/demos/grouping/typeahead-grouping.ts

```diff
@@ -0,0 +1,76 @@
+import {Component, Input} from '@angular/core';
+import {Observable} from 'rxjs';
+import {debounceTime, distinctUntilChanged, map} from 'rxjs/operators';
+
+const states = ['Alabama', 'Alaska', 'American Samoa', 'Arizona', 'Arkansas', 'California', 'Colorado',
+  'Connecticut', 'Delaware', 'District Of Columbia', 'Federated States Of Micronesia', 'Florida', 'Georgia',
+  'Guam', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine',
+  'Marshall Islands', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana',
+  'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota',
+  'Northern Mariana Islands', 'Ohio', 'Oklahoma', 'Oregon', 'Palau', 'Pennsylvania', 'Puerto Rico', 'Rhode Island',
+  'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virgin Islands', 'Virginia',
+  'Washington', 'West Virginia', 'Wisconsin', 'Wyoming'];
+
+@Component({
+  selector: 'ngbd-typeahead-grouping',
+  templateUrl: './typeahead-grouping.html',
+  styles: [`.form-control { width: 300px; }`]
+})
+export class NgbdTypeaheadGrouping {
```

Comment:
I think our US list of states is not relevant for that kind of demo.
We could inspire ourselves from official Twitter Typeahead.js [multi-datasets example](https://twitter.github.io/typeahead.js/examples#multiple-datasets).

Reply:
The demo is also broken if you type 2 letters, like "Co" (Colorado, Connecticut). They will be split in 2 groups

Reply:
My previous comment here is still valid to me, we should think about another problem domain for this demo. List of states is not good...